cloud/amazon/s3_website.py

```diff
@@ -0,0 +1,245 @@
+    required: true
+    default: null 
+  error_key:
+    description:
+      - "The object key name to use when a 4XX class error occurs. To remove an error key, set to None."
+    required: false
+    default: null
+  redirect_all_requests:
+    description:
+      - "Describes the redirect behavior for every request to this s3 bucket website endpoint"
+    required: false
+    default: null
+  region:
+    description:
+     - "AWS region to create the bucket in. If not set then the value of the AWS_REGION and EC2_REGION environment variables are checked, followed by the aws_region and ec2_region settings in the Boto config file.  If none of those are set the region defaults to the S3 Location: US Standard."
+    required: false
+    default: null
+  state:
+    description:
+      - "Add or remove s3 website configuration"
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  suffix:
+    description:
+      - "Suffix that is appended to a request that is for a directory on the website endpoint (e.g. if the suffix is index.html and you make a request to samplebucket/images/ the data that is returned will be for the object with the key name images/index.html). The suffix must not include a slash character."
+    required: false
+    default: index.html
+    
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Configure an s3 bucket to redirect all requests to example.com
+- s3_website:
+    name: mybucket.com
+    redirect_all_requests: example.com
+    state: present
+
+# Remove website configuration from an s3 bucket
+- s3_website:
+    name: mybucket.com
+    state: absent
+    
+# Configure an s3 bucket as a website with index and error pages
+- s3_website:
+    name: mybucket.com
+    suffix: home.htm
+    error_key: errors/404.htm
+    state: present
+    
+'''
+
+try:
+    import boto.ec2
+    from boto.s3.connection import OrdinaryCallingFormat, Location
+    from boto.s3.website import RedirectLocation, RoutingRules, RoutingRule
+    from boto.exception import BotoServerError
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+
+def enable_bucket_as_website(connection, module):
+    
+    bucket_name = module.params.get("name")
+    suffix = module.params.get("suffix")
+    error_key = module.params.get("error_key")
+    if error_key == "None":
+        error_key = None
+    redirect_all_requests = module.params.get("redirect_all_requests")
+    changed = False
+    
+    if redirect_all_requests is not None:
+        redirect_location = RedirectLocation(hostname=redirect_all_requests)
+    else:
+        redirect_location = None
+    
+    try:
+        bucket = connection.get_bucket(bucket_name)
+        if compare_bucket_as_website(bucket, module) is False:
+            bucket.delete_website_configuration()
+            bucket.configure_website(suffix, error_key, redirect_location)
```

Comment:
@njsaunders @wimnat Just tried this myself and it seems that the default value of `index.html` for the `suffix` argument causes the error. Adding `suffix = None` after line 104 fixes it for me.